Sources/_NIOConcurrency/AsyncAwaitSupport.swift

```diff
@@ -38,6 +38,22 @@ extension EventLoopFuture {
     }
 }
 
+extension EventLoopGroup {
+    /// Shuts down the event loop gracefully.
+    @available(macOS 12, iOS 15, tvOS 15, watchOS 8, *)
+    @inlinable
+    public func shutdownGracefully() async throws {
+        return try await withCheckedThrowingContinuation { cont in
+            self.shutdownGracefully { error in
+                if let error = error {
+                    cont.resume(throwing: error)
+                }
+                cont.resume()
```

Comment:
Continuations must be resumed _exactly once_: this will fail in the event we hit the error path, as we'll resume twice.

Reply:
Of course, oops. I forgot the early-return/switch/else here. Total fail 😅.